src/main/java/seedu/trackbeau/storage/JsonAdaptedCustomer.java

```diff
@@ -0,0 +1,171 @@
+
+/**
+ * Jackson-friendly version of {@link Customer}.
+ */
+class JsonAdaptedCustomer {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Person's %s field is missing!";
+
+    private final String name;
+    private final String phone;
+    private final String email;
+    private final String address;
+    private final String skinType;
+    private final String hairType;
+    private final List<JsonAdaptedTag> staffs = new ArrayList<>();
+    private final List<JsonAdaptedTag> services = new ArrayList<>();
+    private final List<JsonAdaptedTag> allergies = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonAdaptedPerson} with the given customer details.
+     */
+    @JsonCreator
+    public JsonAdaptedCustomer(@JsonProperty("name") String name, @JsonProperty("phone") String phone,
+                               @JsonProperty("email") String email, @JsonProperty("address") String address,
+                               @JsonProperty("skinType") String skinType,
+                               @JsonProperty("hairType") String hairType,
+                               @JsonProperty("staffs") List<JsonAdaptedTag> staffs,
+                               @JsonProperty("services") List<JsonAdaptedTag> services,
+                               @JsonProperty("allergies") List<JsonAdaptedTag> allergies) {
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.address = address;
+        this.skinType = skinType;
+        this.hairType = hairType;
+        if (staffs != null) {
+            this.staffs.addAll(staffs);
+        }
+        if (services != null) {
+            this.services.addAll(services);
+        }
+        if (allergies != null) {
+            this.allergies.addAll(allergies);
+        }
+    }
+
+    /**
+     * Converts a given {@code Person} into this class for Jackson use.
+     */
+    public JsonAdaptedCustomer(Customer source) {
+        name = source.getName().fullName;
+        phone = source.getPhone().value;
+        email = source.getEmail().value;
+        address = source.getAddress().value;
+        skinType = source.getSkinType().value;
+        hairType = source.getHairType().value;
+        staffs.addAll(source.getStaffs().stream()
+                .map(JsonAdaptedTag::new)
+                .collect(Collectors.toList()));
+        services.addAll(source.getServices().stream()
+                .map(JsonAdaptedTag::new)
+                .collect(Collectors.toList()));
+        allergies.addAll(source.getAllergies().stream()
+                .map(JsonAdaptedTag::new)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted customer object into the model's {@code Person} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted customer.
+     */
+    public Customer toModelType() throws IllegalValueException {
```

Comment:
This method seems quite long, should we try abstracting the different parts of this method?